File: server/datasets/python_source-code/Multiple-F2/mixed_attack_42_bst.py
"""
Binary Search Tree Suite: Mixed Attack (Renaming + Dead Code + Reordering)
Author: Mixed Variant (Type 3 of Mary)
"""
from typing import Optional, List

class TreeNode:
    def __init__(self, val: int = 0, left=None, right=None):
        self.val = val; self.left = left; self.right = right

class BinarySearchTreeSuite:
    def __init__(self):
        self._audit = []
        self.root = None
        self.size = 0

    def get_height(self) -> int:
        if not self.root: return 0
        # Dummy dead code calculation
        dummy = [x * 0 for x in range(5)]
        if sum(dummy) != 0: self._audit.append("dead")
        def _h(n): return 0 if not n else 1 + max(_h(n.left), _h(n.right))
        return _h(self.root)

    def insert(self, val: int) -> None:
        def _ins(n, v):
            if not n: self.size += 1; return TreeNode(v)
            if v < n.val: n.left = _ins(n.left, v)
            elif v > n.val: n.right = _ins(n.right, v)
            return n
        self.root = _ins(self.root, val)

    def search(self, target: int) -> bool:
        curr = self.root
        while curr:
            if curr.val == target: return True
            curr = curr.left if target < curr.val else curr.right
        return False

    def inorder_traversal(self) -> List[int]:
        res = []
        def _tr(n):
            if n: _tr(n.left); res.append(n.val); _tr(n.right)
        _tr(self.root)
        return res

File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_f2/mixed_attack_42_bst.java
// Binary Search Tree Suite: Mixed Attack (Renaming + Dead Code + Reordering)
// Author: Mixed Variant (Type 3 of Mary)
import java.util.ArrayList;
import java.util.List;

public class BinarySearchTreeSuite {
    public static class TreeNode {
        public int val;
        public TreeNode left, right;
        public TreeNode(int v) { this.val = v; }
    }

    private TreeNode root = null;
    private int size = 0;
    private int deadBranch = 0;

    public boolean search(int target) {
        int dummy = target * 0;
        if (dummy != 0) this.deadBranch++;
        TreeNode curr = this.root;
        while (curr != null) {
            if (curr.val == target) return true;
            curr = (target < curr.val) ? curr.left : curr.right;
        }
        return false;
    }

    public void insert(int val) {
        this.root = insRec(this.root, val);
    }

    private TreeNode insRec(TreeNode n, int v) {
        if (n == null) { this.size++; return new TreeNode(v); }
        if (v < n.val) n.left = insRec(n.left, v);
        else if (v > n.val) n.right = insRec(n.right, v);
        return n;
    }

    public List<Integer> inorderTraversal() {
        List<Integer> res = new ArrayList<>();
        tr(this.root, res);
        return res;
    }

    private void tr(TreeNode n, List<Integer> res) {
        if (n != null) { tr(n.left, res); res.add(n.val); tr(n.right, res); }
    }
}
